pilot/cmd/pilot-agent/main.go

```diff
@@ -168,7 +168,7 @@ var (
 						// only support the default config, or env variable
 						ns = os.Getenv("ISTIO_NAMESPACE")
 						if ns == "" {
-							ns = "istio-system"
+							ns = model.IstioSystemNamespace
```

Comment:
Why ? Agent doesn't read model ( it's a config map in istio-system that is not mounted in pod ns). 
It may be better to remove the use of 'ns' - agent should only need the pilot address, as a parameter, which can
now be provided by annotation or at inject time.

Reply:
You mean we can get the namespace from pilot address for sure?
But I think the 'ns' logic here is for the scenario that  the discovery address here is a remote address.

Reply:
Pilot does not have to be in the namespace, it's enough to be a DNS name, please do not make this assumption.

I agree that hardcode needs to be changed, @costinm I see you added the dependency on the namespace. What is this used for? It has to be removed.

Reply:
@costinm I'd like to post a follow-up PR for the namespace thing.
But forgive me for my poor knowledge on spiffe.  How would a pilotSAN looks like if no namespace specified?
So the SAN does not have to be in the format as it is now? just fine to make it unique/identifiable?
And should the mixerSAN (used in pilot-discovery) be updated as well?